{"name": "test_product_image", "status": "failed", "statusDetails": {"message": "AssertionError: Cant find element by locator: ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')", "trace": "self = <page_objects.ProductPage.ProductPage object at 0x101c0a250>, locator = ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')\n\n    def _verify_element_presence(self, locator: tuple):\n        try:\n>           return WebDriverWait(self.browser, self.browser.t).until(EC.visibility_of_element_located(locator))\n\npage_objects/BasePage.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"e68f7b8c93b3c1f619ba60b1ade0a330\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x101aeaf70>, message = ''\n\n    def until(self, method, message=''):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.time() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except InvalidSelectorException as e:\n                raise e\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, 'screen', None)\n                stacktrace = getattr(exc, 'stacktrace', None)\n            time.sleep(self._poll)\n            if time.time() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       0   chromedriver                        0x0000000100642d98 chromedriver + 4337048\nE       1   chromedriver                        0x000000010063ae14 chromedriver + 4304404\nE       2   chromedriver                        0x0000000100267a5c chromedriver + 293468\nE       3   chromedriver                        0x00000001002acd50 chromedriver + 576848\nE       4   chromedriver                        0x00000001002e7908 chromedriver + 817416\nE       5   chromedriver                        0x00000001002a0a5c chromedriver + 526940\nE       6   chromedriver                        0x00000001002a1908 chromedriver + 530696\nE       7   chromedriver                        0x0000000100608de4 chromedriver + 4099556\nE       8   chromedriver                        0x000000010060d2a0 chromedriver + 4117152\nE       9   chromedriver                        0x000000010061352c chromedriver + 4142380\nE       10  chromedriver                        0x000000010060dda0 chromedriver + 4119968\nE       11  chromedriver                        0x00000001005e5a74 chromedriver + 3955316\nE       12  chromedriver                        0x000000010062aa48 chromedriver + 4237896\nE       13  chromedriver                        0x000000010062abc4 chromedriver + 4238276\nE       14  chromedriver                        0x000000010063aa8c chromedriver + 4303500\nE       15  libsystem_pthread.dylib             0x000000018cafffa8 _pthread_start + 148\nE       16  libsystem_pthread.dylib             0x000000018cafada0 thread_start + 8\n\nvenv/lib/python3.9/site-packages/selenium/webdriver/support/wait.py:89: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nbrowser = <selenium.webdriver.remote.webdriver.WebDriver (session=\"e68f7b8c93b3c1f619ba60b1ade0a330\")>\n\n    def test_product_image(browser):\n        MainPage(browser).go_to_product_page()\n>       image_tap = ProductPage(browser).click_image_tap()\n\ntests/test_opencart.py:96: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npage_objects/ProductPage.py:38: in click_image_tap\n    image_tap = self._element(self.IMAGE_TAP)\npage_objects/BasePage.py:27: in _element\n    return self._verify_element_presence(locator)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <page_objects.ProductPage.ProductPage object at 0x101c0a250>, locator = ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')\n\n    def _verify_element_presence(self, locator: tuple):\n        try:\n            return WebDriverWait(self.browser, self.browser.t).until(EC.visibility_of_element_located(locator))\n        except TimeoutException:\n>           raise AssertionError(\"Cant find element by locator: {}\".format(locator))\nE           AssertionError: Cant find element by locator: ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')\n\npage_objects/BasePage.py:24: AssertionError"}, "start": 1695470691477, "stop": 1695470696940, "uuid": "82238dd2-2f8a-4c9f-9519-ead964061a0c", "historyId": "00e447fed6f0b3fcc93d0081decf7990", "testCaseId": "00e447fed6f0b3fcc93d0081decf7990", "fullName": "tests.test_opencart#test_product_image", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_opencart"}, {"name": "host", "value": "Karinas-MacBook-Air.local"}, {"name": "thread", "value": "63199-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_opencart"}]}